src/Spec2-CommonWidgets/SpSimpleFileEditPresenter.class.st

```diff
@@ -0,0 +1,142 @@
+"
+I define a simple code editor. 
+Before saving I create a backup of the edited file. 
+
+
+Here is a typical use: 
+
+```
+SpSimpleFileEditPresenter new 
+	onFileReference: '/Library/Preferences/pharo/6.0/settings.st' asFileReference;
+	openDialog 
+```
+"
+Class {
+	#name : 'SpSimpleFileEditPresenter',
+	#superclass : 'SpPresenter',
+	#instVars : [
+		'label',
+		'fileReference',
+		'codeText'
+	],
+	#category : 'Spec2-CommonWidgets',
+	#package : 'Spec2-CommonWidgets'
+}
+
+{ #category : 'examples' }
+SpSimpleFileEditPresenter class >> example [
+	<script>
+	
+	| presenter fm |
+	fm := FileSystem memory / 'unknow.st'.
+	fm writeStreamDo: [ :each | each nextPutAll: 'Please edit me and save me!' ].
+	presenter := self new.
+	presenter
+		label: 'The meaning of life?';
+		onFileReference: fm ;
+		openDialog
+]
+
+{ #category : 'layout' }
+SpSimpleFileEditPresenter >> backupFileReference [
+	
+	| basename nameWithoutExtension |
+	basename := fileReference basename.
+	nameWithoutExtension :=  fileReference basenameWithoutExtension.
+	^ fileReference parent / (nameWithoutExtension, 'Old.txt')
+]
+
+{ #category : 'layout' }
+SpSimpleFileEditPresenter >> calculateLabelHeight [
+
+	^ 16
+]
+
+{ #category : 'layout' }
+SpSimpleFileEditPresenter >> defaultLayout [
+
+	^ SpBoxLayout newTopToBottom
+		spacing: 5;
+	  	add: (SpBoxLayout newLeftToRight
+				add: label expand: true;
+				yourself)
+	  		height: self calculateLabelHeight;
```

Comment:
it is not recommanded to hardcode the height in Spec.
Using expand: false on the TopToBottom layout should be enough.
```smalltalk
^ SpBoxLayout newTopToBottom
		spacing: 5;
	  	add: (SpBoxLayout newLeftToRight
				add: label expand: true;
				yourself)
	  		expand: false;
	add: codeText;
	yourself
```